### DISCUSSION TITLE
Use with Unity?

### DISCUSSION THREAD

**Discussion by wiverson** [supabase-community/supabase-csharp#65] [General] [answered]
Is anyone using this library with Unity? Any thoughts/suggestions? One thing I would find very useful is the ability to use the new signInWithIdToken / signInWithApple functionality.

I found this in the gotrue-dart, a C# version would be very nice.

https://github.com/supabase/gotrue-dart/blob/4909d8725540a5b4b0a8d87a72ed69a374dd470c/lib/src/gotrue_client.dart#L226

**Comment by acupofjose** [comment]
I've been messing around with it in Godot - particularly the gotrue-csharp functionality. I haven't used it in Unity, a previous dev said that he had struggled getting realtime-csharp working with the Unity compiler, but I'm not sure what all he tried. This project (https://github.com/eliyap/supabase-unity) was working previously, but is on an old version.

Still a little behind getting all the LW7 functionality worked in. The code is in [gotrue-csharp](https://github.com/supabase-community/gotrue-csharp/blob/master/Gotrue/Client.cs) if you want to submit a PR! They're always welcome.

**Comment by wiverson** [comment, marked as answer]
I got it working in Unity, at least basic connection to a Hello World PostREST JS function pretty easily.

1. Install [NuGetForUnity](https://github.com/GlitchEnzo/NuGetForUnity)
2. Install the [UniTask](https://github.com/Cysharp/UniTask) package via the above to make Unity async work properly
3. Install supabase-csharp

I then added a button and a text field and wired them up to this script, and everything seems to be working fine. I added an FPS checker and a spinning cube and the async seems to be working properly.

I'll dig into the native auth stuff and see if I can sort out a PR. The dart version looks like it has the basics, I'll play around and see what I can sort out.

As an aside, I've been thinking about switching to Godot but TBH I just wasted a bunch of time on Flutter and finally gave up and went back to Unity. Maybe for my next project... :)

```csharp
using System.Collections.Generic;
using System.Threading.Tasks;
using Postgrest.Responses;
using Supabase;
using TMPro;
using UnityEngine;

public class SupabaseStuff : MonoBehaviour {
    private string SupabaseURL = "https://[projectid].supabase.co";

    private string SupabasePublicKey =
        "[SUPABASE_PUBLIC_KEY]";

    public TMP_Text result;

    private static Client _supabase;

    private async void StartSupabase() {
        string url = SupabaseURL;
        string key = SupabasePublicKey;

        if (_supabase == null) {
            _supabase = new Client(url, key);
            await _supabase.InitializeAsync();
        }

        Dictionary<string, object> param = new Dictionary<string, object>();
        param["name"] = "howdy";

        Task<BaseResponse> rpc = _supabase.Rpc("hello_js_as_json", param);

        await rpc;

        if (rpc.IsCompleted) {
            result.text = rpc.Result.Content;
        } else {
            result.text = $"{rpc.Status}";
        }
    }

    public void StartPublic() {
        result.text = "...";
        StartSupabase();
    }
}
```

**Reply by acupofjose** [reply]
@wiverson thanks so much for diving into it and providing an example! I'll admit to having a bit of a dream at seeing these libraries used in some game-dev projects. If you do end up using it, please let me know! I'd love to list your project (open-source or not) in the repo readme!
